plugins/org.eclipse.thym.core/src/org/eclipse/thym/core/config/WidgetModel.java

```diff
@@ -439,6 +451,63 @@ public void modelDirtyStateChanged(IStructuredModel model, boolean isDirty) {
 		}
 	}
 
+	private void updateEnginesFromConfigXML() {
+		final IProject project = configXMLtoIFile().getProject();
+		WorkspaceJob prepareJob = new WorkspaceJob(NLS.bind("Execute cordova prepare for {0}",
+				project.getName())) {
+
+			@Override
+			public IStatus runInWorkspace(IProgressMonitor monitor) throws CoreException {
+				HybridProject hybridProject = HybridProject.getHybridProject(project);
+				if (hybridProject == null) {
+					HybridCore.log(IStatus.WARNING, 
+							"Updating from config.xml: Could not get HybridProject", null);
+					return Status.CANCEL_STATUS;
+				}
+				HybridMobileEngine[] activeEngines = 
+						hybridProject.getActiveEnginesFromPlatformsJson();
+				CordovaCLI cordova = CordovaCLI.newCLIforProject(hybridProject);
+				IStatus status = Status.OK_STATUS;
+				SubMonitor sm = SubMonitor.convert(monitor, 100);
+
+				if (editableWidget != null) {
+					List<Engine> configEngines = editableWidget.getEngines();
+					if (configEngines == null) {
+						return new Status(IStatus.ERROR, HybridCore.PLUGIN_ID,
+								"No engine found in config.xml");
+					}
+					for (Engine e : configEngines) {
+						String platformSpec = e.getName() + "@" + e.getSpec();
+						if (checkPlatformInstalled(activeEngines, e.getName())) {
+							status = cordova.platform(Command.UPDATE, sm.newChild(60), platformSpec)
```

Comment:
When multiple platforms are processed this will cause the status of only the last platform update to be reported. I think we need to report the most severe status . Take a look at `org.eclipse.core.runtime.MultiStatus`